utils.py

```diff
@@ -724,73 +739,94 @@ def Stack():
 
 
 class FIFOQueue(Queue):
```

Comment:
The code here is fine, but I think we can just use `collections.deque`.

Reply:
But then we have to change every uses of `FIFOQueue` and replace its `pop` with `popleft`. I thought the idea behind these classes here was to be able to use `PriorityQueue` and `FIFOQueue` (and maybe `Stack`) with a the same code and just change in the constructor.

Reply:
Yes, @goswami-rahul you are correct that the motivation of a single interface was to make the various queues the same, so that they could be interchangeable, and we could highlight the similarities. This was a priority in the 1st and 2nd edition of the book (and the code goes all the way back to then). But in the 3rd (and soon 4th) edition, we accept that each search function is somewhat different, and so it is ok, and maybe even preferably, to have popleft instead of pop.